### PR TITLE
runner refactor: Model to_runner/to_runnable interface

### PR DESCRIPTION
**Pull request by parano**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by sauyon — APPROVED** [review]
One minor comment but othewise LGTM

File: bentoml/_internal/runner/runner.py [hunk 1]
```diff
@@ -84,10 +83,13 @@
 """
 
 
-@attr.define()
+@attr.define(frozen=True)
 class RunnerMethod:
-    runner: "Runner" = attr.field()
+    runner: Runner = attr.field()
```
[the remark this answers is not in the document]
Comment (bojiang) [reply bentoml/_internal/runner/runner.py:88]: good to know

File: bentoml/_internal/runner/runner.py [hunk 2]
```diff
@@ -179,84 +168,133 @@ async def async_run_method(
...
+            ).with_config_overrides(runner_name),
+            runner_methods=runner_methods,
+            scheduling_strategy=scheduling_strategy,
         )
 
-        for name in runnable_class.get_method_configs().keys():
-            setattr(self, name, RunnerMethod(self, name))
+        for runner_method in self.runner_methods:
+            if runner_method.method_name == "__call__":
+                setattr(self, "run", runner_method.run)
```
Comment (bojiang) [inline bentoml/_internal/runner/runner.py:261]: great!

**Review by bojiang — APPROVED** [review]

**Comment by bojiang** [comment]
merge for now

**Pull request merged**